sahi/model.py

```diff
@@ -663,3 +665,156 @@ def _create_object_prediction_list_from_original_predictions(
+
+        if self.category_mapping is None:
+            category_names = {str(i): COCO_CLASSES[i] for i in range(len(COCO_CLASSES))}
+            self.category_mapping = category_names
+
+    def perform_inference(self, image: np.ndarray, image_size: int = None):
+        """
+        Prediction is performed using self.model and the prediction result is set to self._original_predictions.
+        Args:
+            image: np.ndarray
+                A numpy array that contains the image to be predicted. 3 channel image should be in RGB order.
+            image_size: int
+                Inference input size.
+        """
+        # TO DO: Check image size parameter(dont image_size parameter)
+        from sahi.utils.torchvision import numpy_to_torch, resize_aspect_ratio
+
+        if image_size is not None:
+            model_img = resize_aspect_ratio(image, self.image_size)
+            model_img = numpy_to_torch(model_img)
+            prediction_result = self.model([model_img])
+
+        else:
+            model_img = numpy_to_torch(image)
+            prediction_result = self.model([model_img])
+
+        self._original_predictions = prediction_result
+
+    @property
+    def num_categories(self):
+        """
+        Returns number of categories
+        """
+        return len(self.category_mapping)
+
+    @property
+    def has_mask(self):
+        """
+        Returns if model output contains segmentation mask
+        """
+        return self.model.with_mask
+
+    @property
+    def category_names(self):
+        return self.category_mapping
+
+    def _create_object_prediction_list_from_original_predictions(
+        self,
+        shift_amount_list: Optional[List[List[int]]] = [[0, 0]],
+        full_shape_list: Optional[List[List[int]]] = None,
+    ):
+        """
+        self._original_predictions is converted to a list of prediction.ObjectPrediction and set to
+        self._object_prediction_list_per_image.
+        Args:
+            shift_amount_list: list of list
+                To shift the box and mask predictions from sliced image to full sized image, should
+                be in the form of List[[shift_x, shift_y],[shift_x, shift_y],...]
+            full_shape_list: list of list
+                Size of the full image after shifting, should be in the form of
+                List[[height, width],[height, width],...]
+        """
+        original_predictions = self._original_predictions
+
+        # compatilibty for sahi v0.8.20
+        if isinstance(shift_amount_list[0], int):
+            shift_amount_list = [shift_amount_list]
+        if full_shape_list is not None and isinstance(full_shape_list[0], int):
+            full_shape_list = [full_shape_list]
+
+        # parse boxes, masks, scores, category_ids from predictions
+        from sahi.utils.torchvision import COCO_CLASSES
+
+        prediction_class = [COCO_CLASSES[i] for i in list(original_predictions[0]["labels"].numpy())]
```

Comment:
@kadirnar su sekilde olmali:

```suggestion
        prediction_class = [self.category_mapping[i] for i in list(original_predictions[0]["labels"].numpy())]
```